src/AppInstallerCommonCore/Manifest/ManifestCommon.cpp

```diff
@@ -410,4 +478,48 @@ namespace AppInstaller::Manifest
             return {};
         }
     }
+
+    std::map<DWORD, ExpectedReturnCodeEnum> GetDefaultKnownReturnCodes(InstallerTypeEnum installerType)
+    {
+        switch (installerType)
+        {
+        case InstallerTypeEnum::Burn:
+        case InstallerTypeEnum::Wix:
+        case InstallerTypeEnum::Msi:
+            // See https://docs.microsoft.com/windows/win32/msi/error-codes
+            return
+            {
+                { ERROR_INSTALL_ALREADY_RUNNING, ExpectedReturnCodeEnum::InstallInProgress },
+                { ERROR_DISK_FULL, ExpectedReturnCodeEnum::DiskFull },
+                { ERROR_INSTALL_SERVICE_FAILURE, ExpectedReturnCodeEnum::ContactSupport },
+                { ERROR_SUCCESS_REBOOT_REQUIRED, ExpectedReturnCodeEnum::RebootRequiredToFinish },
+                { ERROR_SUCCESS_REBOOT_INITIATED, ExpectedReturnCodeEnum::RebootInitiated },
+                { ERROR_INSTALL_USEREXIT, ExpectedReturnCodeEnum::CancelledByUser },
+                { ERROR_PRODUCT_VERSION, ExpectedReturnCodeEnum::AlreadyInstalled },
+                { ERROR_INSTALL_REJECTED, ExpectedReturnCodeEnum::BlockedByPolicy },
+            };
+        case InstallerTypeEnum::Inno:
+            // See https://jrsoftware.org/ishelp/index.php?topic=setupexitcodes
+            return
+            {
+                { 2, ExpectedReturnCodeEnum::CancelledByUser },
+                { 5, ExpectedReturnCodeEnum::CancelledByUser },
+                { 8, ExpectedReturnCodeEnum::RebootRequiredForInstall },
+            };
+        case InstallerTypeEnum::Msix:
+            // See https://docs.microsoft.com/en-us/windows/win32/appxpkg/troubleshooting
+            return
+            {
+                { ERROR_INSTALL_PREREQUISITE_FAILED, ExpectedReturnCodeEnum::MissingDependency },

```

Comment:
>ERROR_INSTALL_PREREQUISITE_FAILED [](http://example.com/codeflow?start=18&length=33)

For MSIX apis, the returned error codes are HResults, so you'll have to convert these win 32 error codes to HResults,
otherwise, they may not get matched. Use below in allmsix return codes
HRESULT_FROM_WIN32(ERROR_INSTALL_PREREQUISITE_FAILED)